core/domain/state_domain.py

```diff
@@ -745,16 +758,336 @@ def get_all_html_content_strings(self):
+                    ca_value['content_id'], ca_value['unicode_str'])
+
+            if obj_type == schema_utils.SCHEMA_OBJ_TYPE_SUBTITLED_HTML:
+                return SubtitledHtml(
+                    ca_value['content_id'], ca_value['html'])
+
+        ca_value = InteractionCustomizationArg.traverse_by_schema_and_convert(
+            ca_schema,
+            copy.deepcopy(ca_dict['value']),
+            convert_content_to_domain_obj
+        )
+
+        return cls(ca_value, ca_schema)
+
+    def __init__(self, value, schema):
+        """Initializes a InteractionCustomizationArg domain object.
+
+        Args:
+            value: *. The value of the interaction customization argument.
+            schema: dict. The schema defining the specification of the value.
+        """
+        self.value = value
+        self.schema = schema
+
+    def get_content_ids(self):
+        """Get all content_id's from SubtitledHtml and SubtitledUnicode in the
+        customization argument.
+
+        Returns:
+            list(str). A list of content_id's.
+        """
+        return InteractionCustomizationArg.traverse_by_schema_and_get(
+            self.schema,
+            self.value,
+            ('SubtitledHtml', 'SubtitledUnicode'),
+            lambda x: x.content_id
+        )
+
+    def get_html(self):
+        """Get all html from SubtitledHtml in the customization argument.
+
+        Returns:
+            list(str). A list of html.
+        """
+
+        return InteractionCustomizationArg.traverse_by_schema_and_get(
+            self.schema,
+            self.value,
+            ('SubtitledHtml'),
+            lambda x: x.html
+        )
+
+    @staticmethod
+    def traverse_by_schema_and_convert(schema, value, conversion_fn):
+        """Helper function that recursively traverses an interaction
+        customization argument spec to locate any SubtitledHtml or
+        SubtitledUnicode objects, and applies a conversion function to the
+        customization argument value.
+
+        Args:
+            schema: dict. The customization dict to be modified: dict
+                with a single key, 'value', whose corresponding value is the
+                value of the customization arg.
+            value: dict. The current nested customization argument value to be
+                modified.
+            conversion_fn: function. The function to be used for converting the
+                content. It is passed the customization argument value and
+                obj_type.
+
+        Returns:
+            dict. The converted customization dict.
+        """
+        schema_type = schema['type']
+        schema_obj_type = schema.get('obj_type', None)
+
+        if (schema_obj_type == schema_utils.SCHEMA_OBJ_TYPE_SUBTITLED_UNICODE or
+                schema_obj_type == schema_utils.SCHEMA_OBJ_TYPE_SUBTITLED_HTML):
+            value = conversion_fn(value, schema_obj_type)
+        elif schema_type == schema_utils.SCHEMA_TYPE_LIST:
```

Comment:
Why the duplication with traverse_ca_by_schema_and_convert_html above?

Reply:
Good point - done